framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -376,13 +380,67 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 			this._ccuFrequency.liveness >= newBlockHeader.height - this._lastCertifiedHeight
 		) {
 			// TODO: _createCCU needs to be implemented which will create and send the CCU transaction
+			const listOfCCMs = this._getListOfCCMs(
+				this._chainConnectorState.crossChainMessages,
+				newBlockHeader,
+			);
+			// eslint-disable-next-line no-console
+			console.log(listOfCCMs);
 			await this._createCCU();
 			// if the transaction is successfully submitted then update the last certfied height and do the cleanup
 			// TODO: also check if the state is growing, delete everything from the inMemory state if it goes beyond last 3 rounds
 			this._cleanup();
 		}
 	}
 
+	private _getListOfCCMs(
+		crossChainMessages: CcmWithHeight[],
+		newBlockHeader: BlockHeader,
+	): CcmWithHeight[][] {
+		const listOfCCMs: CcmWithHeight[][] = [];
+		const filteredCCMs = crossChainMessages.filter(
+			cmmWithHeight => cmmWithHeight.height >= newBlockHeader.aggregateCommit.height,
+		);
+		if (filteredCCMs.length === 0) {
+			return listOfCCMs;
+		}
+
+		const getListWithLastIndex = (
+			startIndex: number,
+		): [list: CcmWithHeight[], newIndex: number] => {
+			const list: CcmWithHeight[] = [];
+			let totalSize = 0;
+			let i = startIndex;
+
+			for (; i < filteredCCMs.length; i += 1) {
+				const ccmWithHeight = filteredCCMs[i];
+				const ccmBytes = codec.encode(ccmSchema, ccmWithHeight.ccm);
+				const size = ccmBytes.length;
+				totalSize += size;
+				if (totalSize > CCU_TOTAL_CCM_SIZE) {
+					break;
+				}
+
+				list.push(ccmWithHeight);
+			}
+
+			return [list, i];
+		};
+
+		const getListOfList = (inputList: CcmWithHeight[][], startIndex: number) => {
```

Comment:
```suggestion
		const bundleCCMsBySize = (inputList: CcmWithHeight[][], startIndex: number) => {
```

Reply:
Adding this suggestion because it's not clear from the name alone that we are grouping the CCMs together by `CCU_TOTAL_CCM_SIZE`.